pkg/internal/testhelpers/postgres_container.go

```diff
@@ -432,6 +432,24 @@ func dropTimescaleAndToolkitExtensionsInDB(dbname string) error {
 	return nil
 }
 
+// updateTimescaleDBInDatabase updates the timescaledb version in a named database
+func updateTimescaleDBInDatabase(database string) error {
+	ctx := context.Background()
+	db, err := pgx.Connect(ctx, PgConnectURL(database, Superuser))
+	if err != nil {
+		return fmt.Errorf("could not connect to %s db: %w", database, err)
+	}
+	defer db.Close(ctx)
+
+	_, err = db.Exec(context.Background(), "ALTER EXTENSION timescaledb UPDATE")
```

Comment:
Is this safe to do if the extension is not installed previously? Maybe we need a check for that.

Reply:
It is not safe to do this if the extension is not installed. For our tests, the extension is always installed though.

I did experiment with this (checking if the extension is installed, and conditionally updating it), unfortunately querying a table to check if the extension is installed triggers loading the timescaledb extension, meaning that you can no longer update it in the same connection. At that point I gave up on that approach and went for the simpler, but less robust code.

Reply:
I kind of addressed this in a code comment in the file below:

```
// Note: We don't actually know which databases have timescaledb installed
// in them. Instead of spending time trying to find out, we assume that
// they all do. This is probably mostly correct.
```